wrappers/ios/Sources/Wrapper/Reader/ZXIBarcodeReader.mm

```diff
@@ -0,0 +1,151 @@
+    // default method
+    return [self readCIImage:[[CIImage alloc] initWithCVImageBuffer:pixelBuffer]];
+}
+
+- (NSArray<ZXIResult *> *)readCIImage:(nonnull CIImage *)image {
+    CGImageRef cgImage = [self.ciContext createCGImage:image fromRect:image.extent];
+    auto results = [self readCGImage:cgImage];
+    CGImageRelease(cgImage);
+    return results;
+}
+
+- (NSArray<ZXIResult*> *)readImageView: (ImageView)imageView {
+    Results results = ReadBarcodes(imageView, [ZXIBarcodeReader DecodeHintsFromZXIOptions:self.hints]);
+
+    NSMutableArray* zxiResults = [NSMutableArray array];
+    for (auto result: results) {
+        if(result.status() == DecodeStatus::NoError) {
+            const std::wstring &resultText = result.text();
+            NSString *text = [[NSString alloc] initWithBytes:resultText.data()
+                                                      length:resultText.size() * sizeof(wchar_t)
+                                                    encoding:NSUTF32LittleEndianStringEncoding];
+
+            NSData *bytes = [[NSData alloc] initWithBytes:result.bytes().data() length:result.bytes().size()];
+            [zxiResults addObject:
+             [[ZXIResult alloc] init:text
+                              format:ZXIFormatFromBarcodeFormat(result.format())
+                               bytes:bytes
+                            position:[[ZXIPosition alloc]initWithPosition: result.position()]
+             ]];
+        }
+    }
+    return zxiResults;
+}
+
+- (NSArray<ZXIResult *> *)readCGImage: (nonnull CGImageRef)image {
+    CGColorSpaceRef colorSpace = CGColorSpaceCreateWithName(kCGColorSpaceGenericGray);
+    CGFloat cols = CGImageGetWidth(image);
+    CGFloat rows = CGImageGetHeight(image);
+    NSMutableData *data = [NSMutableData dataWithLength: cols * rows];
+
+
+    CGContextRef contextRef = CGBitmapContextCreate(
+                                                    data.mutableBytes,// Pointer to backing data
+                                                    cols,                      // Width of bitmap
+                                                    rows,                     // Height of bitmap
+                                                    8,                          // Bits per component
+                                                    cols,              // Bytes per row
+                                                    colorSpace,                 // Colorspace
+                                                    kCGBitmapByteOrderDefault); // Bitmap info flags
+    CGContextDrawImage(contextRef, CGRectMake(0, 0, cols, rows), image);
+    CGContextRelease(contextRef);
+
+    ImageView imageView = ImageView(
+              static_cast<const uint8_t *>(data.bytes),
+              static_cast<int>(cols),
+              static_cast<int>(rows),
+              ImageFormat::Lum);
+
+    Results results = ReadBarcodes(imageView, [ZXIBarcodeReader DecodeHintsFromZXIOptions:self.hints]);
```

Comment:
This should call the above `readImageView` thereby removing the following code duplication.

Reply:
Thank you for pointing that out!